openroad.bzl

```diff
@@ -722,3 +730,24 @@ def build_openroad(
             cmd = get_entrypoint_cmd(make_pattern, design_config, stage_config, True, make_targets, docker_image = docker_image),
             outs = outs.get(stage, []),
         )
+
+        # Prepare GUI targets
+        if stage in ("synth", "floorplan", "place", "cts", "route", "final"):
+            base_targets = [Label(target_name + "_" + stage)]
+            if stage == "synth":
+                write_stage_config(
+                    name = target_name + "_gui_" + stage + "_config",
+                    stage = stage,
+                    srcs = stage_cfg_srcs,
+                    stage_args = stage_args["synth"] + stage_args["synth_sdc"] + lefs_args,
+                )
+                base_targets.append(Label(target_name + "_synth_sdc"))
+                stage_config = Label("@@//:" + target_name + "_gui_" + stage + "_config.mk")
+            native.genrule(
+                name = target_name + "_" + stage + "_gui",
+                tools = [Label("//:docker_shell")],
+                srcs = [make_pattern, design_config, stage_config] + base_targets,
+                cmd = "echo \"chmod -R +w . && \"" + get_entrypoint_cmd(make_pattern, design_config, stage_config, Label("//:docker_shell"), "gui_" + stage, docker_image = docker_image) + " > $@ ",
```

Comment:
which folder is this chmod -R +w in? It should be the bazel-bin folder, right?

Reply:
The chmod is changing permissions on the current working folder, it should be bazel-bin.

```
oyvind@small-cigar:~/bazel-orfs$ bazel run L1MetadataArray_full_final_gui
INFO: Invocation ID: 8364ca15-a6c4-4eaf-b6fa-b1738bccb841
INFO: Analyzed target //:L1MetadataArray_full_final_gui (0 packages loaded, 1 target configured).
INFO: Found 1 target...
WARNING: failed to create one or more convenience symlinks for prefix 'bazel-':
  cannot create symbolic link bazel-out -> /home/oyvind/.cache/bazel/_bazel_oyvind/6e078b44989a9ce1500a63165a20fd2a/execroot/_main/bazel-out:  /home/oyvind/bazel-orfs/bazel-out (File exists)
Target //:L1MetadataArray_full_final_gui up-to-date:
  bazel-bin/L1MetadataArray_full_final_gui.sh
INFO: Elapsed time: 0.054s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/L1MetadataArray_full_final_gui.sh
/home/oyvind/.cache/bazel/_bazel_oyvind/6e078b44989a9ce1500a63165a20fd2a/execroot/_main/bazel-out/k8-fastbuild/bin/L1MetadataArray_full_final_gui.sh: line 1: bazel-out/k8-opt-exec-ST-13d3ddad9198/bin/docker_shell: No such file or directory
oyvind@small-cigar:~/bazel-orfs$ cat bazel-bin/L1MetadataArray_full_final_gui.shchmod -R +w . && OR_IMAGE=openroad/flow-ubuntu22.04-builder:latest DESIGN_CONFIG=bazel-out/k8-fastbuild/bin/L1MetadataArray_full_config.mk STAGE_CONFIG=bazel-out/k8-fastbuild/bin/L1MetadataArray_full_final_config.mk MAKE_PATTERN=./final-bazel.mk RULEDIR=bazel-out/k8-fastbuild/bin bazel-out/k8-opt-exec-ST-13d3ddad9198/bin/docker_shell make gui_final
```

Reply:
That was a leftover which should be removed during rebase, now it's gone